### PR TITLE
Add a test for empty key / empty passwords for AESCipher.

### PR DESCRIPTION
**Pull request by phaazon**

### CODE REVIEW

File: core/test/crypto/encryption_test.cpp [hunk 1]
```diff
@@ -154,4 +154,38 @@ TEST(Encryption, EncryptDecryptWithCipherHugeText) {
...
-}
\ No newline at end of file
+}
+
+TEST(Encryption, EncryptDecryptWithCipherEmptiness) {
+    auto rng = std::make_shared<OpenSSLRandomNumberGenerator>();
+
+    std::vector<uint8_t> vec(BIG_TEXT.begin(), BIG_TEXT.end());
+    BytesReader input(vec);
+    BytesWriter encrypted;
```
Comment (teams2ua) [inline core/test/crypto/encryption_test.cpp]: encrtypted can be created inside lambda, so you will not need clear() method on BytesWriter

Comment (phaazon) [reply core/test/crypto/encryption_test.cpp]: While this is true, it will also make the code perform more memory allocations.

Comment (phaazon) [reply core/test/crypto/encryption_test.cpp]: I’ll remove `clear` because it’s just tests and no one seems interested in that function.
